test/junit/scala/collection/immutable/LazyListLazinessTest.scala

```diff
@@ -874,8 +855,15 @@ class LazyListLazinessTest {
       }
     }
 
-    LazyList.range(0, 1000)
```

Comment:
because this was a range of `Int` and not of `CustomLong`, it didn't actually test that `LazyList.range` doesn't evaluate all the elements of the range - evaluating `Int`s doesn't increment `counter`. This fixes it so that it is properly a range of `CustomLong`, as well as adds tests for where the step is not (the equivalent of) 1